src/globus_sdk/_sphinxext.py

```diff
@@ -170,7 +174,96 @@ def gen_rst(self):
         yield ""
 
 
+class ExternalDocLink(AddContentDirective):
+    has_content = False
+    required_arguments = 1
+    optional_arguments = 0
+    # allow for spaces in the argument string
+    final_argument_whitespace = True
+    option_spec = {"mode": directives.unchanged, "ref": directives.unchanged_required}
+
+    def gen_rst(self):
+        message = self.arguments[0].strip()
+
+        mode = "default"
+        if "mode" in self.options:
+            mode = self.options["mode"]
+
+        if mode == "default":
+            base_url = "https://docs.globus.org/api"
+            relative_link = self.options["ref"]
+
+            yield (
+                f"See `{message} <{base_url}/{relative_link}>`_ in the "
+                "API documentation for details."
+            )
+        elif mode in (
+            "groups",
+            "gcs",
+        ):
+            raise ValueError(f"Unsupported extdoclink mode {mode}. TODO: add support")
+        else:
+            raise ValueError(f"Unsupported extdoclink mode {mode}")
+
+
+class ExpandTestingFixture(AddContentDirective):
+    has_content = False
+    required_arguments = 1
+    optional_arguments = 0
+    option_spec = {
+        "case": directives.unchanged,
+    }
+
+    def gen_rst(self):
+        from globus_sdk._testing import get_response_set
+
+        response_set_name = self.arguments[0]
+        casename = "default"
+        if "case" in self.options:
+            casename = self.options["case"].strip()
+        response_set = get_response_set(response_set_name)
+        response = response_set.lookup(casename)
+        if response.json is not None:
+            yield ".. code-block:: json"
+            yield ""
+            output_lines = json.dumps(
+                response.json, indent=2, separators=(",", ": ")
+            ).split("\n")
+            for line in output_lines:
+                yield f"    {line}"
+        elif response.body is not None:
+            yield ".. code-block::"
+            yield ""
+            output_lines = response.body.split("\n")
+            for line in output_lines:
+                yield f"    {line}"
+        else:
+            raise RuntimeError(
+                "Error loading example content for response "
+                f"{response_set_name}:{casename}. Neither JSON nor text body was found."
+            )
+
+
+class PaginatedUsage(AddContentDirective):
+    has_content = False
+    required_arguments = 1
+    optional_arguments = 0
+
+    def gen_rst(self):
+        yield "This method supports paginated access. "
+        yield "To use the paginated variant, give the same arguments as normal, "
+        yield "but prefix the method name with ``paginated``, as in"
+        yield ""
+        yield f">>> client.paginated.{self.arguments[0]}(...)"
```

Comment:
Should this be wrapped in a code block?

```suggestion
        yield ".. code-block:: pycon"
        yield ""
        yield f"    >>> client.paginated.{self.arguments[0]}(...)"
```